webapp/components/test-file-results.js

```diff
@@ -80,7 +83,65 @@ class TestFileResults extends TestRunsUIQuery(
   }
 
   static get observers() {
-    return ['fetchTestFile(path, testRuns)'];
+    return ['loadData(path, testRuns, structuredSearch)'];
+  }
+
+  async loadData(path, testRuns, structuredSearch) {
+    // Run a search query, including subtests, as well as fetching the results file.
+    let [searchResults, resultsTable] = await Promise.all([
+      this.fetchSearchResults(path, testRuns, structuredSearch),
+      this.fetchTestFile(path, testRuns),
+    ]);
+
+    if (resultsTable && searchResults) {
+      const test = searchResults.results.find(r => r.test === path);
+      if (test) {
+        let subtests = new Set(test.subtests);
+        let [first, ...others] = resultsTable;
+        const matches = others.filter(t => subtests.has(t.name));
+        resultsTable = [first, ...matches];
+      }
+    }
+    this.resultsTable = resultsTable;
+  }
+
+  async fetchSearchResults(path, testRuns, structuredSearch) {
+    if (!testRuns || !testRuns.length || !this.structuredQueries || !structuredSearch) {
+      return;
+    }
+
+    // Combine the query with " and [path]".
+    let q = {
+      and: [
+        {pattern: path},
```

Comment:
Will something terrible happen if this matches more than one test? It could happen with variants, where both `/some/test.html?a` and `/some/test.html?aa` could exist.